src/code/fault.c

```diff
@@ -273,31 +341,42 @@ void Fault_Sleep(u32 duration) {
     Fault_SleepImpl(duration);
 }
 
+void PadMgr_RequestPadData(Input* input, s32 mode);
+
 void Fault_PadCallback(Input* input) {
-    //! @bug This function is not called correctly and thus will crash from reading a bad pointer at 0x800C7E4C
+    //! @bug This function is not called correctly, it is missing a leading PadMgr* argument
     PadMgr_RequestPadData(input, 0);
 }
 
-void Fault_UpdatePadImpl() {
-    sFaultStructPtr->padCallback(&sFaultStructPtr->padInput);
+void Fault_UpdatePadImpl(void) {
+    sFaultInstance->padCallback(&sFaultInstance->padInput);
 }
 
-u32 Fault_WaitForInputImpl() {
-    Input* input = &sFaultStructPtr->padInput;
+/**
+ * Awaits user input
+ *
+ * L toggles auto-scroll
+ * DPad-Up enables osSyncPrintf output
+ * DPad-Down disables osSyncPrintf output
+ * A and DPad-Right continues and returns true
+ * DPad-Left continues and returns false
+ */
+u32 Fault_WaitForInputImpl(void) {
+    Input* input = &sFaultInstance->padInput;
     s32 count = 600;
     u32 kDown;
 
     while (true) {
-        Fault_Sleep(0x10);
+        Fault_Sleep(16);
```

Comment:
I think all the `Fault_Sleep(16);` are
```suggestion
        Fault_Sleep(1000 / 60);
```
but is it truly better than `16` idk